### ISSUE TITLE
Cannot use `TemplateSpatialModel` as spatial model in `TSMapEstimator`

### ISSUE THREAD

**Issue by ArmelleJB** [gammapy/gammapy#4920]
**Gammapy version**
gammapy 1.1

**Bug description**
Trying to use a template instead of a gaussian or point source to make significance map with `TSMapEstimator` - the position of the template exists, but there is an error for setting the position (no setter) :
```
model.spatial_model.position = geom.center_skydir
[...]
AttributeError: can't set attribute
```

**Expected behavior**
Should work as `GaussianSpatialModel` but a `position` function is redefined that maybe overwrite the inherited one. Adding
```
@position.setter
    def position(self, skycoord):
        """Spatial model center position"""
        coord = skycoord.transform_to(self.frame)
        self.lon_0.quantity = coord.data.lon
        self.lat_0.quantity = coord.data.lat
```
to the `TemplateSpatialModel` may solve the problem. 

**To Reproduce**
```
ts_spatial_model = TemplateSpatialModel(image)
ts_spectral_model = PowerLawSpectralModel(
    index=2.4, amplitude="1e-11 cm-2 s-1 TeV-1", reference="1 TeV"
)

model_ts = SkyModel(
    spatial_model=ts_spatial_model,
    spectral_model=ts_spectral_model,
    name="model-ts",
)

estimator = TSMapEstimator(
    model=model_ts,
    energy_edges=[1.0,100] * u.TeV,
)

dataset.models=bkg_model
result = estimator.run(dataset)
print(result)
result["sqrt_ts"].plot(add_cbar=True, stretch="linear")
```

**Comment by QRemy** [comment]
One solution could be to create internally a new map, shifting the geom center to the new position.

**Comment by adonath** [comment]
@QRemy I think we can actually just add positional parameters to the `TemplateSpatialModel`. It is a well defined use case to fit the position of a template as well.

**Comment by QRemy** [comment]
> @QRemy I think we can actually just add positional parameters to the `TemplateSpatialModel`. It is a well defined use case to fit the position of a template as well.

I agree if frozen but default.

**Comment by QRemy** [comment]
> > @QRemy I think we can actually just add positional parameters to the `TemplateSpatialModel`. It is a well defined use case to fit the position of a template as well.
> 
> I agree if frozen but default.

**Comment by AtreyeeS** [comment]
Resolved for v1.2 in https://github.com/gammapy/gammapy/pull/4973
